### ISSUE TITLE
Provide a way to pipe client request to "pub serve web" during development

### ISSUE THREAD

**Issue by nicolasgarnier** [ForceUniverse/dart-forcemvc#23]
Typically the Force MVC server serves the built client files located in **/build/web/**. These are usually just JS files and not .dart files since they don't get copied when doing `pub build` (unless using pub build in dev mode which is slow).
To allow for convenient development without running `pub build --mode=dev` each time I noticed dart force will also serve the client dart files located in **/web/** by re-writing the file path (remove /build). This helps a lot for development.

However, since the files are served directly the transformers specified in **pubspec.yaml** are not applied. For example if a developer uses polymer there is a transformer that will apply some modifications to the code. Here is a **pubspec.yaml** for polymer enabled projects:

```
dependencies:
  polymer: ">=0.15.0 <0.16.0"
transformers:
- polymer
```

It's fine if the developer runs `pub build --mode=dev` but it is not convenient as it is very slow and would need to be ran at every modification of the code.

The way I propose we fix this is by allowing developers to run `pub serve` and fetch the client files through the pub server instead of reading them on disk. Because `pub serve` is especially made for development it has several advantages:
- `pub serve` will apply transformers at run time
- A dart2js transformer is added by `pub serve` so you also get the compiled .js files at run time without running `pub build`

Basically you'll get both Javascript and dart files served as if they were compiled but without running a lengthy `pub build`.

I propose the way developers would run this is something like:

```
pub serve web  --hostname 0.0.0.0 --port 7777 &&
export DART_PUB_SERVE=0.0.0.0:7777 &&
dart bin/server.dart
```

In the Force MVC server you just have to check if the environment variable `DART_PUB_SERVE` exists and in that case read the client files from that URI using an HTTP request instead of reading the file on disk. That way you will serve the transformed dart and js file while still serving static files and controllers through your existing code.
